util/query.go

```diff
@@ -29,3 +39,31 @@ func IntQueryParam(ctx echo.Context, name string, defaultValue, minValue, maxVal
 
 	return value, nil
 }
+
+func StringQueryParam(ctx echo.Context, name string, defaultValue string) (string, error) {
+
+	// Get the query parameter value.
+	valueStr := ctx.QueryParam(name)
+	if valueStr == "" {
+		return defaultValue, nil
+	}
+
+	return valueStr, nil
+}
+
+func VerifyDateParameters(ctx echo.Context) (string, string, error) {
+	currentTime := time.Now()
+	oneWeekAgo := time.Now().AddDate(0, 0, -7)
+
+	startDate, err := StringQueryParam(ctx, "startDate", oneWeekAgo.Format(dateFormat))
+	if err != nil {
+		return "", "", ctx.JSON(http.StatusBadRequest, ErrorResponse{Description: err.Error()})
+	}
+
+	endDate, err := StringQueryParam(ctx, "endDate", currentTime.Format(dateFormat))
+	if err != nil{
+		return "", "", ctx.JSON(http.StatusBadRequest, ErrorResponse{Description: err.Error()})
+	}
```

Comment:
There's a bug here where services that call this function won't stop processing after the error is detected in most cases. The reason is that `ctx.JSON` only returns an error if the response can't be sent (which should only happen if TCP connection has been severed). In that case, this function will return two empty strings and `nil`. The result is that the service will continue processing and most likely encounter another error when it tries to query the database with empty strings for timestamp arguments.

The fix here is simply to return the error instead of calling `ctx.JSON`. (That is, `return "", "", err`.) The only catch is that the function calling `VerifyDateParameters` has to check the error return parameter. At least in the case of `BuildAppsHandler`, it does. I'll try to mention any calls to this function that don't.